tests/scancode/test_ignore_files.py

```diff
@@ -68,49 +68,49 @@ def test_ignore_glob_file(self):
         )
         assert is_ignored(*test)
 
-    def test_resource_paths_with_single_file(self):
-        test_dir = self.extract_test_tar('ignore/user.tgz')
-        expected = [
-            'user',
-            'user/ignore.doc',
-            'user/src',
-            'user/src/ignore.doc',
-            'user/src/test',
-            'user/src/test/sample.txt'
-        ]
-        test = [rel_path for abs_path, rel_path in resource_paths(test_dir,{'sample.doc': 'test ignore'})]
-        assert expected == sorted(test)
+    # def test_resource_paths_with_single_file(self):
```

Comment:
I need to add updated tests for `resource_paths` since that function's signature in modified. I was waiting for your go for the abstract class approach :P